packages/atomic/src/components/common/interface/bindings.ts

```diff
@@ -47,6 +47,20 @@ export interface CommonBindings<
    * A reference to the atomic interface element.
    */
   interfaceElement: InterfaceElement;
+
+  /**
+   * Creates a Style Information element with the nonce attribute if it exists.
+   * Note: the element is **not** appended to the DOM.
+   * @returns a style element.
+   */
+  createStyleElement: () => HTMLStyleElement;
```

Comment:
Other alternatives to add these function would be:

* Utility function on the `store`, which can be specific (ie: not the "common" store)
* `@Method()` on the search/insight interface itself

I think these might be better instead of adding them to the "base" interface, in order to limit its scope/narrow the surface a bit.

So, for a given type of `atomic-xyz-interface` component, we can decide to add support for it when necessary, and not force it for all types of `atomic-xyz-interface`.

Reply:
The JSDoc would also greatly benefit from MDN links and/or wording that explain what a nonce is, what type of error it could help solve, etc.

To give an example, if an implementer sees `CSP` directive error from their web browser console, as was shown in the discuss post, they might be tempted to simply copy paste the error message in our documentation. It would be neat if this documentation would pop up correctly.

Reply:
What are some examples of situations in which an implementer may use this feature?

Reply:
> @Method() on the search/insight interface itself

I don't think so, as those methods are potentially required on any component (I think digging up to get the atomic-interface and its method would be a bit crude).

> Utility function on the store, which can be specific (ie: not the "common" store)

I thought about that, but it'd denature the store nature of the store a bit, which is why I went with the bindings.

> We can decide to add support for it when necessary, and not force it for all types of atomic-xyz-interface.

This is a good point, I'll see what I can do to reduce the sprawl and target only the main interface.

Reply:
> What are some examples of situations in which an implementer may use this feature?

The most detailed one is on [Discuss](https://discuss.coveo.com/t/support-00111577-using-coveo-atomic-with-content-security-policy-csp-without-unsafe-inline/11103)

Reply:
> I don't think so, as those methods are potentially required on any component (I think digging up to get the atomic-interface and its method would be a bit crude).

There should not be any digging necessary 🤔 The search interface reference is already passed in bindings to all components

Reply:
Split into NonceBindings